mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoSequenceGenerator.java

```diff
@@ -57,6 +57,9 @@ private void initIndex() {
         coll.createIndex(keys, options);
     }
 
+    // true when sequance collection is initialized and indexes created
+    private static boolean initialized = false;
```

Comment:
There can be more than one collection used for this, so same thing used in lock should be used here.